platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformInvocationFactory.java

```diff
@@ -92,7 +96,25 @@ public Deferrable<Try<ImmutableList<File>>> createInvocation(
                 fileSystemAccess,
                 workspaceServices
             );
+        } else if (!transform.requiresInputChanges()) {
+            // Non-incremental project artifact transforms also run in an immutable workspace
+            workspaceServices = immutableWorkspaceProvider;
```

Comment:
🤔 This also means that we'd use the in-memory cache for non-incremental project transforms. Is that a problem?

Reply:
Why would it be a problem? Because of the additional memory usage?